synapse/api/errors.py

```diff
@@ -225,11 +225,20 @@ def __init__(self, msg="Not found", errcode=Codes.NOT_FOUND):
 
 class AuthError(SynapseError):
     """An error raised when there was a problem authorising an event."""
-
     def __init__(self, *args, **kwargs):
         if "errcode" not in kwargs:
             kwargs["errcode"] = Codes.FORBIDDEN
-        super(AuthError, self).__init__(*args, **kwargs)
+        self.admin_uri = kwargs.get('admin_uri')
+        self.msg = kwargs.get('msg')
+        self.errcode = kwargs.get('errcode')
+        super(AuthError, self).__init__(*args, errcode=kwargs["errcode"])
```

Comment:
So this constructor now looks odd. We should either pass through `**kwargs` to the super constructor or take `errcode` as a parameter instead of `*args, **kwargs`. 

Its also worth noting that `self.msg` and `self.errocde` are set by `SynapseError` constructor, so we don't need to set them here.

I'd probably write this as something like:

```python
    def __init__(self, code, msg, errcode=Codes.FORBIDDEN, admin_url=None):
        self.admin_uri = admin_url
        super(AuthError, self).__init__(code, msg, errcode=errcode)
```

Reply:
Sorry, that's brain dead on my part, thanks for the clean up